drv/sprot-api/src/lib.rs

```diff
@@ -333,232 +422,238 @@ impl From<u8> for MsgType {
     }
 }
 
-#[derive(Copy, Clone, Serialize, Deserialize, SerializedSize)]
-#[repr(C, packed)]
-struct MsgHeader {
-    protocol: Protocol,
-    msgtype: MsgType,
-    len: u16,
+/// A builder/serializer for messages that wraps the transmit buffer
+///
+/// Each public method returns the serialized buffer that can be sent on the
+/// wire.
+pub struct TxMsg {
+    buf: [u8; BUF_SIZE],
 }
 
-#[derive(
-    Copy, Clone, Eq, PartialEq, Serialize, Deserialize, SerializedSize,
-)]
-#[repr(C)]
-pub enum UpdateRspKind {
-    Ok = 1,
-    Value = 2,
-    Error = 3,
-    Unknown = 0xff,
-}
+impl TxMsg {
+    pub fn new() -> TxMsg {
+        TxMsg { buf: [0; BUF_SIZE] }
```

Comment:
This puts a relatively large buffer on the stack, I think? Large buffers in other tasks often end up with `&'static [u8; N]` references to move them to flash (where we have more tooling to keep track of size requirements than we do stack size); e.g., https://github.com/oxidecomputer/hubris/blob/master/task/net/src/buf.rs, https://github.com/oxidecomputer/hubris/blob/master/task/control-plane-agent/src/main.rs#L234-L240.

Reply:
The buffer was already on the stack. I just wrapped it in this type. Presumably this is necessary for performance in the hot path. @lzrd any thoughts here?